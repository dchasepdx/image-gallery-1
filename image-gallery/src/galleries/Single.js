import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Single extends Component {
    constructor(props) {
        super(props);

        this.state = {
            index: 0,

        }

    }

    static PropTypes = {
        images:PropTypes.array.isRequired
    };
    //Two ways to use onClick: 
    //1:  onClick={this.previousBunny}....changes the context of this so must bind even in own component
    //2.   onClick={() =>  this.nextBunny(index)} > no need to bind because arrow function binds lexically//
    nextBunny(index) {
        const arrayLength = this.props.images.length;
        index++;
        if (index > arrayLength -1) {
            index = 0;
        }
        this.setState({ index });
    }

    previousBunny(index) {
        const arrayLenth = this.props.images.length;
        index--;
        if (index < 0) {
            index = arrayLenth - 1;
        }
        this.setState({ index });
    }



    render() {
        const { index } = this.state;
        const image = this.props.images[index];


        if (!this.props.images.length) {
            return null;
        }
        return (
            <div>  
                <span className='nextBunny' style={{
                    fontSize: '50px',
                    cursor: 'pointer'

                }}
                    onClick={() => this.previousBunny(index)}>
                    &larr;
                </span>

                <span className='previousBunny' style={{
                    fontSize: '50px',
                    cursor: 'pointer',
                }}
                    onClick={() => this.nextBunny(index)} >
                    &rarr;
                </span>

                <div>
                    <img className='images' src={image.url} alt='' />
                    <p>{image.title}</p>
                    <p>{image.description} </p>
                </div>

                
            </div>
        )
    }
}