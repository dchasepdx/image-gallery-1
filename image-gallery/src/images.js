import images from './imagesData';

export default {
    get() { return Promise.resolve(images) }
}