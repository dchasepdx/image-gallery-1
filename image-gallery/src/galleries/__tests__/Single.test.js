import React from 'react';
import ReactDOM from 'react-dom';
import Single from '../Single';
import renderer from 'react-test-renderer';
import images from '../../imagesData';



it('renders List view without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Single  images={images}/>, div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<Single images={images}/>).toJSON();
  expect(tree).toMatchSnapshot();
});