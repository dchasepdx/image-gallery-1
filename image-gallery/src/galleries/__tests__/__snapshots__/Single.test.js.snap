// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <span
    className="nextBunny"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "50px",
      }
    }
  >
    ←
  </span>
  <span
    className="previousBunny"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "50px",
      }
    }
  >
    →
  </span>
  <div>
    <img
      alt=""
      className="images"
      src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
    />
    <p>
      Cute Bunny
    </p>
    <p>
      Isn't it fuzzy-wuzzy cutest thing you've ever seen?
       
    </p>
  </div>
</div>
`;